src/Apps/Open3DViewer/Open3DViewer_mac.mm

```diff
@@ -79,6 +79,19 @@ - (BOOL)application:(NSApplication *)sender openFile:(NSString *)filename {
     mOpenEmptyWindow = false;  // LoadAndCreateWindow() always opens a window
     return (LoadAndCreateWindow(filename.UTF8String));
 }
+
+- (void)applicationWillTerminate:(NSNotification *)notification {
+    // The app will terminate after this function exits. This will result
+    // in the Application object in main() getting destructed, but it still
+    // thinks it is running. So tell it to quit.
+    open3d::gui::Application::GetInstance().Quit();
+    // Process events for a while. (Everything should happen in the first
+    // call, but call it multiple times just in case.)
+    int  i  = 0;
+    while (open3d::gui::Application::GetInstance().RunOneTick() && i < 20) {
```

Comment:
This is mostly paranoia. I assume that if you post to the SDL event loop and then read from the event loop that it would show up immediately. But in case it doesn't, loop through a few times. When the app has finished, RunOneTick() will return false and the loop will exit. On my machine it exits on i=0 (which is what I would expect).  There's not a threading issue and nothing with Filament, it is just that we need to make sure that the run loop actually processes the cleanup events. We can't stop and cleanup in the ordinary way because the cleanup involves closing windows, and all of that assumes that events are being processed. We could write special-purpose code to do it outside of the event loop, but I think that's even worse. The problem here is that SDL isn't really using Apple's event loop, so we have to exit our event loop before Apple exits theirs, which they seem to do immediately when the `-applicationWillTerminate:` function returns.
   I could do `while(RunOneTick());` if you'd prefer, although there's the risk that it would never return. (Unlikely, but I tend to assume the worst, hence the 20, just as a safeguard) I could also name the 20 if you'd prefer.